lexos/static/js/scripts_statistics.js

```diff
@@ -1,66 +1,247 @@
+ */
+function submissionError () {
+  if ($('#num_active_files').val() < 1) { return 'You must have at least 1 active documents to proceed!' } else { return null }
+}
 
+/**
+ * The function to convert the form into json
+ * @returns {{string: string}}: the form converted to json
+ */
+function jsonifyForm () {
+  const form = {}
+  $.each($('form').serializeArray(), function (i, field) {
+    form[field.name] = field.value || ''
+  })
+  return form
+}
 
-  // Reset the maximum number of documents when a checkbox is clicked
-  $('.minifilepreview').click(function () {
-    $('#cullnumber').attr('max', $('.minifilepreview:checked').length)
+/**
+ * Send the ajax request.
+ * @param url {string}: the url to post.
+ * @param form {{string: string}}: the form data packed into an object.
+ * @returns {jQuery.Ajax}: an jQuery Ajax object.
+ */
+function sendAjaxRequest (url, form) {
+  return $.ajax({
+    type: 'POST',
+    url: url,
+    contentType: 'application/json; charset=utf-8',
+    data: JSON.stringify(form)
   })
+}
+
+/**
+ * Format the ajax call response to HTML format string.
+ * @param response {json}: a json format string.
+ * @return {string}: formatted file report.
+ */
+function formatFileReportResponse (response) {
+  // Extract constant result from response.
+  const token_name = response['token_name']
+  const mean = `<p>Average document size is ${response['mean']} ${token_name}</p>`
+  const std_deviation = `<p>Standard deviation of documents is ${response['std_deviation']} ${token_name}</p>`
+  const inter_quartile_range = `<p>Inter quartile range of documents is ${response['inter_quartile_range']} ${token_name}</p>`
+  // Extract standard deviation anomaly information.
+  let anomaly_se
+  if (response['anomaly_se'].every(function (element) { return element === null})) {
+    anomaly_se = `<p><b>No</b> anomaly detected by standard deviation test.</p>`
+  } else {
+    anomaly_se = `<p>Anomaly <b>detected</b> by standard error test.</p>`
+    response['anomaly_se'].forEach(function (element) {
+      if (element !== null) { anomaly_se += `<p style="padding-left: 20px">${element.slice(0, 5).bold()}${element.slice(5, element.length)}</p>` }
+    })
+  }
+
+  // Extract inter quartile range anomaly information.
+  let anomaly_iqr
+  if (response['anomaly_iqr'].every(function (element) { return element === null})) {
+    anomaly_iqr = `<p><b>No</b> anomaly detected by inter quartile range test.</p>`
+  } else {
+    anomaly_iqr = `<p>Anomaly <b>detected</b> by inter quartile range test.</p>`
+    response['anomaly_iqr'].forEach(function (element) {
+      if (element !== null) { anomaly_iqr += `<p style="padding-left: 20px">${element.slice(0, 5).bold()}${element.slice(5, element.length)}</p>` }
+    })
+  }
+
+  // Return the retracted information.
+  return `${mean}${std_deviation}${inter_quartile_range}${anomaly_se}${anomaly_iqr}`
```

Comment:
Can you explain to me what are you doing here. because this data is a simple string and unstructured, therefore it is hard to get the information from this unstructured string. What is the reason behind this?